test/SlowTests/Sharding/Issues/RavenDB_16276.cs

```diff
@@ -0,0 +1,148 @@
+
+        using (var store = GetDocumentStore(options))
+        {
+            using (var session = store.OpenSession())
+            {
+                session.Advanced.Patch<Company, string>("companies/1", x => x.Name, "HR");
+                session.Advanced.Patch<Order, string>("orders/1$companies/1", x => x.Employee, "E1");
+
+                session.SaveChanges();
+            }
+        }
+    }
+
+    [RavenTheory(RavenTestCategory.Sharding)]
+    [RavenData(DatabaseMode = RavenDatabaseMode.Sharded)]
+    public void SingleBucket_BatchBehavior_Will_Throw_For_Commands_In_Different_Bucket_Patch(Options options)
+    {
+        options.ModifyDocumentStore = s => s.Conventions.Sharding.BatchBehavior = ShardedBatchBehavior.TransactionalSingleBucketOnly;
+
+        using (var store = GetDocumentStore(options))
+        {
+            using (var session = store.OpenSession())
+            {
+                session.Advanced.Patch<Company, string>("companies/1", x => x.Name, "HR");
+                session.Advanced.Patch<Order, string>("orders/1", x => x.Employee, "E1");
+
+                Assert.Throws<ShardedBatchBehaviorViolationException>(() => session.SaveChanges());
+            }
+        }
+    }
+
+    [RavenTheory(RavenTestCategory.Sharding)]
+    [RavenData(DatabaseMode = RavenDatabaseMode.Sharded)]
+    public void SingleBucket_BatchBehavior_Will_Work_For_Commands_In_Same_Bucket_PatchBatch(Options options)
+    {
+        options.ModifyDocumentStore = s => s.Conventions.Sharding.BatchBehavior = ShardedBatchBehavior.TransactionalSingleBucketOnly;
+
+        using (var store = GetDocumentStore(options))
+        {
+            using (var session = store.OpenSession())
+            {
+                var ids = new List<string>
+                {
+                    "companies/1",
+                    "orders/1$companies/1"
+                };
+
+                session.Advanced.Defer(new BatchPatchCommandData(ids, new PatchRequest { Script = "" }, null));
+
+                session.SaveChanges();
+            }
+        }
+    }
+
+    [RavenTheory(RavenTestCategory.Sharding)]
+    [RavenData(DatabaseMode = RavenDatabaseMode.Sharded)]
+    public void SingleBucket_BatchBehavior_Will_Throw_For_Commands_In_Different_Bucket_PatchBatch(Options options)
+    {
+        options.ModifyDocumentStore = s => s.Conventions.Sharding.BatchBehavior = ShardedBatchBehavior.TransactionalSingleBucketOnly;
+
+        using (var store = GetDocumentStore(options))
+        {
+            using (var session = store.OpenSession())
+            {
+                var ids = new List<string>
+                {
+                    "companies/1",
+                    "orders/1"
+                };
+
+                session.Advanced.Defer(new BatchPatchCommandData(ids, new PatchRequest { Script = "" }, null));
+
+                Assert.Throws<ShardedBatchBehaviorViolationException>(() => session.SaveChanges());
+            }
+        }
+    }
```

Comment:
let's add a test with a cluster-tx.
I expect that if we use cluster-tx it all should succeed, even if this setting is set to single bucket.